presto-hive/src/main/java/io/prestosql/plugin/hive/util/InternalHiveSplitFactory.java

```diff
@@ -69,6 +74,8 @@ public InternalHiveSplitFactory(
             Properties schema,
             List<HivePartitionKey> partitionKeys,
             TupleDomain<HiveColumnHandle> effectivePredicate,
+            Supplier<TupleDomain<ColumnHandle>> dynamicFilterSupplier,
```

Comment:
Let's use https://docs.oracle.com/javase/8/docs/api/java/util/function/BooleanSupplier.html instead. You don't need to pass pair here

Reply:
thanks, it's more convenient to use `BooleanSupplier`